File: 1347-minimum-number-of-steps-to-make-two-strings-anagram/1347-minimum-number-of-steps-to-make-two-strings-anagram.java
class Solution {
    public int minSteps(String s, String t) {
        int n = s.length();
        Map<Character, Integer> map_s = new HashMap<>();
        Map<Character, Integer> map_t = new HashMap<>();

        for (int i=0; i<n; i++) {
            char cs = s.charAt(i);
            char ct = t.charAt(i);

            map_s.put(cs, map_s.getOrDefault(cs, 0) + 1);
            map_t.put(ct, map_t.getOrDefault(ct, 0) + 1);
        }
        int answer = 0;
        
        for (Map.Entry<Character, Integer> e: map_t.entrySet()){
            char c = e.getKey();
            int cnt_t = e.getValue();
            int cnt_s = map_s.getOrDefault(c, 0);
            
            if (cnt_t > cnt_s) {
                answer += (cnt_t - cnt_s);
            }
        }
        return answer;

    }
}

File: 1347-minimum-number-of-steps-to-make-two-strings-anagram/1347-minimum-number-of-steps-to-make-two-strings-anagram.py
class Solution(object):
    def minSteps(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: int
        """
        map_s = {}
        map_t = {}

        for c in s:
            map_s[c] = map_s.get(c,0) + 1
        
        for c in t:
            map_t[c] = map_t.get(c,0) + 1

        answer = 0
        for c, cnt in map_t.items():
            if c not in map_s:
                answer += cnt
            elif cnt > map_s[c]:
                answer += abs(cnt - map_s[c])
        return answer

        